pkg/server/server.go

```diff
@@ -102,3 +114,18 @@ func New(configuration options.Configuration) (controllerruntime.Manager, error)
 
 	return mgr, nil
 }
+
+func k8sCommunicationCheck(client *discovery.DiscoveryClient) (err error) {
+	// retry 6 times with a delay of 5 seconds each retry
+	for range [6]int{} {
+		_, err := client.RESTClient().Get().AbsPath("/api").DoRaw(context.TODO())
```

Comment:
Can we try `_, err =` instead of `_, err :=`? In other words avoid [variable shadowing](https://thanos.io/tip/contributing/coding-style-guide.md/#avoid-variable-shadowing).

Reply:
thanks for pointing this out, will fix this